go/libraries/doltcore/doltdb/foreign_key_coll.go

```diff
@@ -61,18 +61,41 @@ func (f *ForeignKeyViolationError) Error() string {
 		}
 		key, _ := f.ViolationRows[i].NomsMapKey(f.Schema).Value(context.Background())
 		val, _ := f.ViolationRows[i].NomsMapValue(f.Schema).Value(context.Background())
-		valSlice, _ := val.(types.Tuple).AsSlice()
-		all, _ := key.(types.Tuple).Append(valSlice...)
-		str, _ := types.EncodedValue(context.Background(), all)
+
+		str := getRowValuesAsString(key, val)
 		sb.WriteRune('\n')
 		sb.WriteString(str)
 	}
+
+	var refCols []string
+	for _, tct := range f.ForeignKey.TableColumns {
+		col, _ := f.Schema.GetAllCols().GetByTag(tct)
+		refCols = append(refCols, col.Name)
+	}
+
+	fKeyErrorMsg := fmt.Sprintf("a foreign key constraint fails (CONSTRAINT `%s` FOREIGN KEY (`%s`) REFERENCES `%s` (`%s`)) on:%s", f.ForeignKey.Name, strings.Join(refCols, ", "), f.ForeignKey.ReferencedTableName, f.ForeignKey.ReferencedTableIndex, sb.String())
+
 	if terminatedEarly {
-		return fmt.Sprintf("foreign key violations on `%s`.`%s`:%s\n%d more violations are not being displayed",
-			f.ForeignKey.Name, f.ForeignKey.TableName, sb.String(), len(f.ViolationRows)-earlyTerminationLimit)
+		return fmt.Sprintf("%s\n%d more violations are not being displayed",
+			fKeyErrorMsg, len(f.ViolationRows)-earlyTerminationLimit)
 	} else {
-		return fmt.Sprintf("foreign key violations on `%s`.`%s`:%s", f.ForeignKey.Name, f.ForeignKey.TableName, sb.String())
+		return fKeyErrorMsg
+	}
+}
+
+// getRowValuesAsString returns row values as string format.
+// Variables 'key' and 'val' have both 'tags and values' added in them, so tags are removed
+func getRowValuesAsString(key types.Value, val types.Value) string {
+	valSlice, _ := val.(types.Tuple).AsSlice()
```

Comment:
Rather than use `AsSlice` here, we should use `IterFields` instead, and just do one loop over the key and another over the value.